internal/xds/translator/listener.go

```diff
@@ -254,3 +257,65 @@ func buildXdsDownstreamTLSSecret(listenerName string,
 		},
 	}, nil
 }
+
+func buildXdsUDPListener(clusterName string, udpListener *ir.UDPListener) (*listener.Listener, error) {
+	if udpListener == nil {
+		return nil, errors.New("udp listener is nil")
+	}
+
+	statPrefix := "udp"
+
+	route := &udp.Route{
+		Cluster: clusterName,
+	}
+	routeAny, err := anypb.New(route)
+	if err != nil {
+		return nil, err
+	}
+
+	udpProxy := &udp.UdpProxyConfig{
```

Comment:
thoughts on keeping this simpler ?
```
              - name: envoy.filters.udp_listener.udp_proxy
                typed_config:
                  '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.UdpProxyConfig
                  stat_prefix: service
                  cluster: service_udp
                  upstream_socket_config:
                    max_rx_datagram_size: 9000
   ```

Reply:
[Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto#envoy-v3-api-msg-extensions-filters-udp-udp-proxy-v3-udpproxyconfig) in favor of matcher since cluster has been deprecated.

Reply:
ah good point, lets keep this newer API then :)